v2/common/src/main/java/com/google/cloud/teleport/v2/transforms/CsvConverters.java

```diff
@@ -0,0 +1,540 @@
+            break;
+
+          case "INTEGER":
+            writer.value(Integer.parseInt(values.get(i).toString()));
+            break;
+
+          case "SHORT":
+            writer.value(Short.parseShort(values.get(i).toString()));
+            break;
+
+          case "BYTE":
+            writer.value(Byte.parseByte(values.get(i).toString()));
+            break;
+
+          case "FLOAT":
+            writer.value(Float.parseFloat(values.get(i).toString()));
+            break;
+
+          case "TEXT":
+          case "KEYWORD":
+          case "STRING":
+            writer.value((String) values.get(i));
+            break;
+
+          default:
+            throw new RuntimeException("Invalid data type");
+        }
+      }
+      writer.endObject();
+      writer.close();
+      return stringWriter.toString();
+    } else if (headers != null) {
+      writer.beginObject();
+      for (int i = 0; i < headers.size(); i++) {
+        writer.name(headers.get(i));
+        writer.value(values.get(i));
+      }
+      writer.endObject();
+      writer.close();
+      return stringWriter.toString();
+    } else {
+      throw new RuntimeException("No headers or schema specified");
+    }
+  }
+
+  /**
+   * Gets Csv format accoring to <a
+   * href="https://javadoc.io/doc/org.apache.commons/commons-csv">Apache Commons CSV</a>. If user passed invalid format
+   * error is thrown.
+   */
+  public static CSVFormat getCsvFormat(String formatString, @Nullable String delimiter) {
+    CSVFormat format;
+    switch (formatString) {
+      case "Default":
+        format = CSVFormat.DEFAULT;
+        break;
+      case "Excel":
+        format = CSVFormat.EXCEL;
+        break;
+      case "InformixUnload":
+        format = CSVFormat.INFORMIX_UNLOAD;
+        break;
+      case "InformixUnloadCsv":
+        format = CSVFormat.INFORMIX_UNLOAD_CSV;
+        break;
+      case "MongoDBCsv":
+        format = CSVFormat.MONGODB_CSV;
+        break;
+      case "MongoDBTsv":
+        format = CSVFormat.MONGODB_TSV;
+        break;
+      case "MySQL":
+        format = CSVFormat.MYSQL;
+        break;
+      case "Oracle":
+        format = CSVFormat.ORACLE;
+        break;
+      case "PostgreSQLCsv":
+        format = CSVFormat.POSTGRESQL_CSV;
+        break;
+      case "PostgreSQLText":
+        format = CSVFormat.POSTGRESQL_TEXT;
+        break;
+      case "RFC4180":
+        format = CSVFormat.RFC4180;
+        break;
+      case "TDF":
+        format = CSVFormat.TDF;
+        break;
+      default:
+        throw new IllegalArgumentException(
+            "CSV format specified is invalid. Got: "
+                + formatString
+                + "\nExpected one of: "
+                + Arrays.toString(CSVFormat.Predefined.values()));
+    }
+
+    // If a delimiter has been passed and does not match the delimiter found in the standard, set it
+    // here.
+    if (delimiter != null && (delimiter.charAt(0) != format.getDelimiter())) {
```

Comment:
Not sure why "delimiter.charAt(0) != format.getDelimiter()" is needed? if delimiter != null then perhaps we should just set:
format.withDelimiter(delimiter.charAt(0))?

Reply:
Updated to set delimiter by default.